[SHK-6] Member 회원가입 기능 구현

BnDC opened the pull request:
### ⛏ 작업 사항
- Member Entity 구현
- 배송정보 Entity 구현
- 회원가입 기능 구현
- 회원가입 기능 테스트
- Spring Security 기본 filter 설정
- image table full_path 60에서 255으로 길이 조정

### 📝 작업 요약
- 회원가입 기능 구현 및 테스트

### 💡 관련 이슈
- Resolved : SHK-6

BnDC committed:
Feat : [SHK-120] 멤버 Entity 구현

BnDC committed:
Feat : [SHK-121] 배송정보 Entity 구현

BnDC committed:
Feat : [SHK-122] 회원가입 기능 구현

BnDC committed:
Test : [SHK-123] 회원가입 기능 Test Code

BnDC committed:
Test : [SHK-124] Spring Security 기본 Filter 설정

BnDC committed:
Chore : image table full_path 최대길이 60에서 255로 수정

goseungwon commented on src/main/java/com/prgrms/kream/domain/member/controller/MemberController.java:
```diff
@@ -1,8 +1,32 @@
-		return 1;
+	private final MemberFacade memberFacade;
+
+	@PostMapping("")
```
```suggestion
	@PostMapping
```
이렇게 바꾸면 더 보기 좋을 것 같아요 :)

BnDC replied:
수정했습니다!

goseungwon commented on src/main/java/com/prgrms/kream/domain/member/dto/MemberRegisterResponse.java:
```diff
@@ -0,0 +1,4 @@
+package com.prgrms.kream.domain.member.dto;
+
+public record MemberRegisterResponse(Long id) {
+}
```
NIT) 필드가 하나면 레코드를 사용하지 않아도 좋을 것 같아요 🤔

goseungwon commented on src/main/java/com/prgrms/kream/domain/member/facade/MemberFacade.java:
```diff
@@ -1,4 +1,20 @@
+@Service
+@RequiredArgsConstructor
 public class MemberFacade {
+	private final MemberService memberService;
```
하나의 서비스만 참조하는데 Facade를 사용하신 이유가 궁금합니다 :)

BnDC replied:
지금은 없지만 나중에 ImageService를 참조할 기능 구현을 위해서 Facade 도입했습니다.

goseungwon replied:
이부분은 위의 NIT와 비슷한 느낌 같아요
저는 필요할 때 구성하면 되지 않을까? 라는 생각도 있습니다

goseungwon reviewed (commented):
안녕하세요 고승원 입니다. 🌞
코멘트 남길 부분이 거의 없네요!
확인 부탁드립니다~!

Kim-Changgyu commented on src/main/java/com/prgrms/kream/common/mapper/MemberMapper.java:
```diff
@@ -1,4 +1,19 @@
+import com.prgrms.kream.domain.member.model.Member;
+
 public class MemberMapper {
+
```
정적 메서드만 갖는 DTO - Entity 간 변환 클래스라면 생성자는 제한 해보는 방식은 어떻게 생각하세요 !?

BnDC replied:
수정했습니다!

Kim-Changgyu commented on src/main/java/com/prgrms/kream/domain/member/model/Member.java:
```diff
@@ -1,4 +1,60 @@
+	@Column(name = "email", nullable = false, unique = true, length = 20)
+	private String email;
+
+	@Embedded
+	private Phone phone;
+
+	@Embedded
+	private Password password;
```
Embedded 쓰니까 Entity가 한 눈에 확 들어오네요 👍

Kim-Changgyu reviewed (approved):
안녕하세요 스크럼 마스터님
특별하게 코멘트할 부분은 없네요 !
고생하셨습니다 :) 👍👍

BnDC committed:
Style : naver-checkstyle-rules에 맞게 수정

BnDC committed:
Style : MemberMapper method name 수정

- memberRegisterRequestToMember를 toMember로 수정

BnDC committed:
Refactor : MemberMapper 기본 생성자 차단

BnDC committed:
Style : 불필요한 문자열 제거

goseungwon reviewed (commented):
빠른 피드백 반영 좋습니다!! 💪

BnDC committed:
Fix : MemberRegisterRequest bean validation 추가

- name 길이 제한 20자 validation 추가
- eamil 공백, 길이 제한 20자 validation 추가

BnDC committed:
Fix : register @Valid 추가

BnDC committed:
Chore : dto package 수정

- request package 추가 및 MemberRegisterRequest 이동
- resposne package 추가 및 MemberRegisterResponse 이동
- MemberDto 삭제
